crates/sequencer/src/serde/contract.rs

```diff
@@ -0,0 +1,246 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+use starknet_api::core::EntryPointSelector;
+use starknet_api::deprecated_contract_class::{EntryPoint, EntryPointOffset, EntryPointType};
+
+use super::program::SerializableProgram;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
can you annotate which code is taken from katana? for credit but also to know which code is technically reviewed, tested and works in katana currently

Reply:
Makes sense, completed via [this](https://github.com/kkrt-labs/ef-tests/pull/159/commits/d7001aacb5a993ccb1621910a171bd4efb25b9ef) commit.